controllers/flinkcluster_updater.go

```diff
@@ -610,6 +616,20 @@ func (updater *ClusterStatusUpdater) deriveClusterStatus(
 	}
 	status.Control = controlStatus
 
+	// handle job update
```

Comment:
Can we refactor this logic into a state machine style? Making states explicitly for better readability.

Reply:
Done.